workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -449,15 +449,18 @@ def _apply_postphysics_to_dycore_state(self) -> Diagnostics:
             if self._postphysics_only_diagnostic_ml:
                 rename_diagnostics(diagnostics)
             else:
-                updated_state = add_tendency(self._state, tendency, dt=self._timestep)
-                updated_state[TOTAL_PRECIP] = precipitation_sum(
+                updated_state_from_tendency = add_tendency(
+                    self._state, tendency, dt=self._timestep
+                )
+                updated_state_from_tendency[TOTAL_PRECIP] = precipitation_sum(
                     self._state[TOTAL_PRECIP],
                     diagnostics[self._postphysics_stepper.net_moistening],
                     self._timestep,
                 )
-                diagnostics[TOTAL_PRECIP] = updated_state[TOTAL_PRECIP]
+                diagnostics[TOTAL_PRECIP] = updated_state_from_tendency[TOTAL_PRECIP]
+                self._state.update_mass_conserving(updated_state_from_tendency)
+                updated_state = assign_attrs_from(self._state, self._state_updates)
```

Comment:
This fix would be more explicit if it only assigned the units rather than other attrs. I would also consider pushing this logic into `self._state.update_mass_conserving` so we don't need to copy paste it again.

Reply:
Honestly, `Quantity` should not require `unit` IMO. It is just a landmine to step on with no tangible benefit.

Reply:
Refactored to `self._state.update_mass_conserving`